accord-core/src/main/java/accord/coordinate/CoordinationFailed.java

```diff
@@ -61,4 +70,21 @@ void set(TxnId txnId, RoutingKey homeKey)
     {
         return homeKey;
     }
+
+    /**
+     * Wrap the exception without changing the type so asynchronous callbacks can add their own stack
+     */
+    public CoordinationFailed wrap()
+    {
+        checkState(this.getClass() == CoordinationFailed.class);
+        return new CoordinationFailed(txnId, homeKey, this);
+    }
+
+    public static Throwable wrap(Throwable t)
+    {
+        if (t instanceof CoordinationFailed)
```

Comment:
not safe... we almost never use `CoordinationFailed` directly, we use an extending type... so the type check will match many things, then `wrap()` will fail as we *require* the top level class to be `CoordinationFailed` (which is almost never true)